test/test_torch.py

```diff
@@ -8334,6 +8334,100 @@ def test_contiguous(self, device):
+
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @onlyOnCPUAndCUDA
+    @dtypes(torch.float, torch.double, torch.half, torch.int, torch.long, torch.cfloat, torch.cdouble, torch.bool, torch.uint8)
+    def test_tensor_split_indices(self, device, dtype):
+        input_sizes = [
+            (0,),
+            (10,),
+            (10, 0),
+            (0, 10),
+            (4, 10),
+            (12, 3),
+        ]
+        indices_args = [
+            (),
+            (0,),
+            (3,),
+            (10,),
+            (-1,),
+            (-10,),
+            (2, -1),
+            (3, 4, 10),
+            (0, -1, 0, 10),
+        ]
+        for input_size in input_sizes:
+            if dtype.is_floating_point or dtype.is_complex:
+                a = torch.randn(input_size, dtype=dtype, device=device)
+            else:
+                if dtype == torch.bool:
+                    a = torch.randint(2, input_size, dtype=dtype, device=device)
+                else:
+                    a = torch.randint(0, 127, input_size, dtype=dtype, device=device)
+            a_n = a.cpu().numpy()
+            for dim in range(a.dim()):
+                for indices in indices_args:
+                    result = torch.tensor_split(a, indices, dim)
+                    msg = f'input_size {input_size}, indices {indices}, dim {dim}'
+                    for result_item in result:
+                        self.assertEqual(result_item.device, torch.device(device), msg=msg)
+                        self.assertEqual(result_item.dtype, dtype, msg=msg)
+                    result_n = np.array_split(a_n, indices, dim)
+                    self.assertEqual(result_n, result, msg=msg)
+
+    @onlyOnCPUAndCUDA
+    def test_tensor_split_errors(self, device):
+        S = 10
+        test_cases = [
+            # input size, sections or indices, dim, error type, error message, numpy error type
+            [(S,), 10, 1, RuntimeError, r'expected dim to be less than self.dim()', IndexError],
+            [(), 10, 0, RuntimeError, r'expected at least a 1-dimensional tensor', IndexError],
+            [(S,), (10,), 1, RuntimeError, r'expected dim to be less than self.dim()', IndexError],
+            [(), (10,), 0, RuntimeError, r'expected at least a 1-dimensional tensor', IndexError],
+            [(S,), 0, 0, RuntimeError, r'number of sections must be larger than 0, got 0', ValueError],
+            [(S,), -1, 0, RuntimeError, r'number of sections must be larger than 0, got -1', ValueError],
+        ]
+        for input_size, sections_or_indices, dim, err, err_msg, numpy_err in test_cases:
+            a = torch.randn(input_size, device=device)
+            msg = f'input_size {input_size}, sections_or_indices {sections_or_indices}, dim {dim}'
+            with self.assertRaisesRegex(err, err_msg, msg=msg):
+                torch.tensor_split(a, sections_or_indices, dim)
+            if TEST_NUMPY:
```

Comment:
Good news: we actually import numpy always now, so we don't need these checks anymore